source/fundamentals/monitoring.txt

```diff
@@ -0,0 +1,406 @@
+- :java-docs:`MongoClient <apidocs/mongodb-driver-sync/com/mongodb/client/MongoClient.html>`
+- :java-docs:`ConnectionCheckedOutEvent <apidocs/mongodb-driver-core/com/mongodb/event/ConnectionCheckedOutEvent.html>`
+- :java-docs:`ConnectionCheckOutFailedEvent <apidocs/mongodb-driver-core/com/mongodb/event/ConnectionCheckOutFailedEvent.html>`
+
+.. _monitoring-jmx:
+
+Monitor Connection Pools with JMX
+---------------------------------
+
+You can monitor connection pool events using **Java Management Extensions (JMX)**. 
+JMX provides tools to monitor applications and devices.
+
+For more information on JMX, see 
+`the official Oracle JMX documentation <https://docs.oracle.com/javase/tutorial/jmx/index.html>`__. 
+
+JMX Support in the MongoDB Driver
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To enable JMX connection pool monitoring, add an instance of the
+``JMXConnectionPoolListener`` class to your ``MongoClient``.
+
+The ``JMXConnectionPoolListener`` class performs the following actions: 
+
+- Creates MXBean instances for each ``mongod`` or ``mongos`` process the driver
+  maintains a connection pool with.
+- Registers these MXBean instances with the platform MBean server. 
+
+MXBeans registered on the platform MBean server have the following properties:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Property
+     - Description
+
+   * - ``clusterId``
+     - A client-generated unique identifier. This identifier ensures that
+       each MXBean the driver makes has a unique name when an application has 
+       multiple ``MongoClient`` instances connected to the same MongoDB deployment.
+
+   * - ``host``
+     - The hostname of the machine running the ``mongod`` or ``mongos`` process.
+
+   * - ``port``
+     - The port on which the ``mongod`` or ``mongos`` process is listening.
+
+   * - ``minSize``
+     - The minimum size of the connection pool, including idle and in-use connections.
+
+   * - ``maxSize``
+     - The maximum size of the connection pool, including idle and in-use connections.
+
+   * - ``size``
+     - The current size of the connection pool, including idle and in-use connections.
+
+   * - ``checkedOutCount``
+     - The current count of connections that are in use.
+
+
+All MXBean instances created by the driver are under the domain
+``"org.mongodb.driver"``.
+
+For more information on the topics discussed in this subsection, see the
+following resources:
+
+- `Oracle Platform MBean Server Reference documentation <https://docs.oracle.com/en/java/javase/16/management/overview-java-se-monitoring-and-management.html#GUID-F7B9AB8A-F5A8-472A-AEC6-93B5B7FBE7CE>`__
+- `Oracle MXBean documentation <https://docs.oracle.com/javase/tutorial/jmx/mbeans/mxbeans.html>`__
+
+JMX and JConsole Example
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example shows how you can monitor the driver's connection pools using JMX
+and **JConsole**. JConsole is a JMX compliant GUI monitoring tool that comes with
+the Java Platform.
```

Comment:
Question: Is there a reason to include JConsole here? If so, I think there needs to be more information around why it's useful and when to use it before jumping into an example.

Reply:
I believe some discussion of JConsole is necessary on this page. JMX requires a JMX compliant monitoring tool, the easiest to use JMX compliant tool [(and the tool referenced by Oracle in it's JMX monitoring guide ](https://docs.oracle.com/en/java/javase/16/management/monitoring-and-management-using-jmx-technology.html#GUID-E17343B0-FED2-4050-88D7-E4793E704ED5)) is JConsole. For this reason, I believe JConsole must be mentioned.

Reply:
I agree I think this makes sense. I have expanded the disclaimer to also include JMX and moved it to the top of the example.

Fix: